src/stan/lang/grammars/statement_grammar_def.hpp

```diff
@@ -124,8 +124,8 @@ namespace stan {
         | increment_log_prob_statement_r(_r1)       // key "increment_log_prob"
         | increment_target_statement_r(_r1)         // key "target"
         | for_statement_r(_r1)                      // key "for"
-        | for_array_statement_r(_r1)                // key "for"
         | for_matrix_statement_r(_r1)               // key "for"
+        | for_array_statement_r(_r1)                // key "for"
```

Comment:
Maybe that earlier code was right which insisted on an array---they're different statements here, so they probably have separate validation.